### PR TITLE
Array task

### PR DESCRIPTION
**Pull request by dreyhiflden**

### CODE REVIEW

File: capital_first.js [hunk 1]
```diff
@@ -1,5 +1,18 @@
-function capitalsFirst(str){
+function capitalsFirst(str) {
+  const checkCase = char => char === char.toUpperCase() ? 1 : 2;
+  const regEx = /[a-z]/i;
 
+  return str
+    .split(' ')
+    .filter(v => regEx.test(v[0]))
```
Comment (carrot-e) [inline capital_first.js]: You created a variable `regEx` and used it only once. This doesn't make much sense. You could rewrite:  
`.filter(v => /[a-z]/i.test(v[0])`

File: capital_first.js [hunk 2]
```diff
@@ -1,5 +1,18 @@
-function capitalsFirst(str){
+function capitalsFirst(str) {
```
Comment (carrot-e) [inline capital_first.js]: :+1:

File: double_trouble.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+function trouble(x, t) {
+  for (let i=0; i < x.length; i++) {
```
Comment (carrot-e) [inline double_trouble.js]: Check codestyle.

File: morse.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  z: '--..'
+};
+
+let transformToMorse = (word) => {
+  return word
+    .split('')
+    .reduce((acc, cur) => acc + object[cur], '');
+};
+
+let uniqueMorseRepresentations = function(words) {
```
Comment (carrot-e) [inline morse.js]: Why not arrow function (like transformToMorse)?

Comment (carrot-e) [inline morse.js]: Well done.

File: capital_first.js (hunk 1 above)
Comment (dreyhiflden) [reply capital_first.js]: Now I’ll know that if we use something once, there’s no need to put it in a variable. Thanks.

File: double_trouble.js (hunk 3 above)
Comment (dreyhiflden) [reply double_trouble.js]: Fixed codestyle.

File: morse.js (hunk 4 above)
Comment (dreyhiflden) [reply morse.js]: Changed to my favorite arrow function

File: morse.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  let arr = [];
+
+  for (let i = 0; i < words.length; i++) {
+    arr.push(transformToMorse(words[i]));
+  }
+
+  return arr.filter((v, i) => arr.indexOf(v) === i).length;
+};
+
+console.log(uniqueMorseRepresentations(["gin", "zen", "gig", "msg"]));
```
Comment (carrot-e) [inline morse.js]: Single quotes!

**Pull request merged**